src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue

```diff
@@ -985,35 +986,35 @@ export default {
 		},
 	},
 	methods: {
-		verifyOrApplyPromotion() {
+		async verifyOrApplyPromotion() {
```

Comment:
Using the async/await flow here will ensure we awat the promo application or getting promo info from the basket before any other steps.